cylc/flow/platforms.py

```diff
@@ -248,10 +248,13 @@ def get_platform_from_group(
     TODO: Uses host_selection methods; should also allow custom select methods.
     """
     if bad_hosts:
-        good_platforms = set()
+        good_platforms = []
         for platform in group['platforms']:
-            if set(platform_from_name(platform)['hosts']) - bad_hosts:
-                good_platforms.add(platform)
+            if any(
+                host for host in platform_from_name(platform)['hosts']
+                if host not in bad_hosts
```

Comment:
When I think about it, the use of `any()` may sometimes be advantageous in terms of performance due to short-circuiting.

One suggestion to simplify
```suggestion
                host not in bad_hosts
                for host in platform_from_name(platform)['hosts']
```